multinet/api/tests/conftest.py

```diff
@@ -65,12 +95,50 @@ def populated_edge_table(owned_workspace: Workspace, populated_node_table: Table
 
 
 @pytest.fixture
-def populated_network(owned_workspace: Workspace, populated_edge_table: Table) -> Network:
+def populated_network(readable_workspace: Workspace, populated_edge_table: Table) -> Network:
+    node_tables = list(populated_edge_table.find_referenced_node_tables().keys())
+    network_name = Faker().pystr()
+    return Network.create_with_edge_definition(
+        name=network_name,
+        workspace=readable_workspace,
+        edge_table=populated_edge_table.name,
+        node_tables=node_tables,
+    )
+
+
+@pytest.fixture
+def public_populated_network(public_workspace: Workspace, populated_edge_table: Table) -> Network:
```

Comment:
Instead of creating fixtures for the combination of populated/public/private tables, networks, etc., I think we could create a couple util functions in `conftest.py` with the following signature:

```python
def populated_table(workspace: Workspace, edge: bool) -> Table:
def populated_network(workspace: Workspace) -> Network: ​
```

`populated_table` would basically be a combination of the existing `populated_node_table` and `populated_edge_table` fixtures, with an if statement. `populated_network` would use `populated_table` function to create the node and edge tables it needs, then creating and returning a network that uses them.

We could also remove the need for the `writeable_workspace`, `readable_workspace`, etc. fixtures (maybe keeping `public_workspace`), and just call `workspace.set_user_permission(user, WorkspacePermission.<perm>)` for whatever test needs it. This would keep us from making excessive fixtures, while allowing us to cover every case.

How does this sound to you?

Reply:
That sounds good to me. I figured that all of the fixtures I was creating were clunky. I'll work on cleaning this up.

Reply:
I actually think we can keep `populated_network` and `populated_<node/edge>_table` as fixtures. I'd just need to give each test access to the `workspace` and `user` fixtures and set permissions there. I think its cleaner to keep them as fixtures. Writing utility functions to generate them that we call from within tests feels less "best practice" to me. I think keeping them as fixtures is easier to understand as well.

Reply:
The problem with keeping them as fixtures is that they're now tied to the permissions that the user has on the workspace. So to cover any case, we'd need to have the following fixtures:

* `public_populated_node_table`
* `public_populated_edge_table`
* `public_populated_network`
* `private_populated_node_table`
* `private_populated_edge_table`
* `private_populated_network`
* `readable_populated_node_table`
* `readable_populated_edge_table`
* `readable_populated_network`
* `writeable_populated_node_table`
* `writeable_populated_edge_table`
* `writeable_populated_network`
* `maintainable_populated_node_table`
* `maintainable_populated_edge_table`
* `maintainable_populated_network`
* `owned_populated_node_table`
* `owned_populated_edge_table`
* `owned_populated_network`

This seems silly since for each entity (node table, edge table, network), we'll have 6 fixtures that are all identical, except for the permission level of the workspace fixture they take as an argument. To me, this feels much less like best practice than having a function which populates the given workspace fixture with the data we need.

Reply:
This is in response to a previously deleted comment where I suggested keeping fixtures for `populated_network`, `populated_edge_table`, and `populated_node_table`

Reply:
Agreed on avoiding combinatorial explosion here.

If the non-explodey way feels non-best-practice, then some consultation with #girder4 might be useful to see what other advice may be applicable here.